### ISSUE TITLE
Enshrouded (1203620)

### ISSUE THREAD

**Issue by pollux78** [ValveSoftware/Proton#7442]
# Compatibility Report
- Name of the game with compatibility issues: Enshrouded
- Steam AppID of the game: 1203620

## System Information
- GPU: RX 6700
- Video driver version: mesa 23.3.3
- Kernel version: 6.7.1
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/pollux56/0922d4590850f8453c082952a261e701
- Proton version: proton experimental (bleeding edge)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1203620.log](https://github.com/ValveSoftware/Proton/files/14047864/steam-1203620.log)

## Symptoms 
try to join a game or create a game and it will crash the desktop when the intro starts

## Reproduction
use proton stable or experimental, try to create game and get past the intro, game will crash your desktop

**Comment by rhoot** [comment]
ProtonDB has a lot of reports of people having the same issue (all with AMD cards), and most not being able to run the game at all with the RADV drivers. I've been able to play just fine with RADV drivers however. I don't have AMDVLK even installed though, so that might be the difference?

```
"EndeavourOS Linux" (64 bit)
Kernel Version: 6.7.1-zen1-1-zen
X Window Manager: wlroots wm
Steam Runtime Version: steam-runtime_0.20231127.68515
Driver: AMD AMD Radeon RX 6900 XT (radeonsi, navi21, LLVM 16.0.6, DRM 3.56, 6.7.1-zen1-1-zen)
Driver Version: 4.6 (Compatibility Profile) Mesa 23.3.3-arch1.1
```


Two problems I did run into with this game:
- When closing the game Steam will sometimes start validating 8.4 GiB of files. Not sure what files or why.
- I was unable to launch the game at all earlier. Not sure what caused it or why, just no window would show up. It looked as if it stalled on launch. Running with `PROTON_LOG=1` gave me nothing but the below. The only solution I could find was to delete the compatdata folder for the game.
```
======================
Proton: 1705700124 proton-8.0-5c
SteamGameId: 1203620
Command: ['/home/johan/games/SteamLibrary/steamapps/common/Enshrouded/enshrouded.exe']
Options: {'forcelgadd'}
depot: 0.20231211.70175
pressure-vessel: 0.20231208.0 scout
scripts: 0.20231208.0
sniper: 0.20231211.70175 sniper 0.20231211.70175
Kernel: Linux 6.7.1-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Sun, 21 Jan 2024 22:13:51 +0000 x86_64
Language: LC_ALL None, LC_MESSAGES None, LC_CTYPE None
Effective WINEDEBUG: +timestamp,+pid,+tid,+seh,+unwind,+threadname,+debugstr,+loaddll,+mscoree
======================
```

Edited to add: The game ID is 1203620

**Comment by pollux78** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7442#issuecomment-1909589488

i dont have vlk installed either and im running endeavour os with the latest packages installed, the only difference im seeing is that your running a wm while im running kde plasma 5.27.10 and i have tried both kwin x11 and kwin wayland with the same results of it crashing when entering the intro of creating a game

**Comment by pollux78** [comment]
[enshrouded.log](https://github.com/ValveSoftware/Proton/files/14049392/enshrouded.log)
here is the enshrouded crash log aswell

**Comment by pollux78** [comment]
![Screenshot_20240126_120958](https://github.com/ValveSoftware/Proton/assets/81181814/2fd4b6e2-d2c7-485a-b7a1-fa419517b3b2)
thats odd the game is working on kde neon unstable with flatpak steam 

edit: if i set reflections to quality the game will crash, setting everything to normal/balanced doesnt make the game crash and its working on endeavour os now so yay :)

**Comment by SolarLune** [comment]
On a fully-updated EndeavourOS, with the following stats (and skipping Steam's shader precaching):

```
GPU: AMD Radeon RX 6650 XT
CPU: AMD Ryzen 5 1600 x12
DE: Gnome 45.3
Kernel: Linux 6.7.1-arch1-1
GPU Driver:  amdgpu
Proton: 8.0-5
```

The game starts and appears to preprocess the shaders. 

However, the problems start once the Keen logo shows up, followed by the main menu. 

Under X11, when the window is not selected, the game is choppy, but appears correctly (which seems to be expected behavior from my memory of playing on Windows). When I mouse over the window, or click on the window to make it active, though, the window turns black. I can click a menu element (Play or Settings, for example) a frame or two before it goes black, but if I then make the window unselected to see the current state, it will appear back at the main menu. This is the same in either Windowed mode or Fullscreen mode (toggled with Alt+Enter).

Under Wayland instead, it's the same as X11 under windowed mode. However, if I switch in fullscreen mode with Alt+Enter, then the game wholly appears correctly, regardless of if the window is active or not. I still get a black screen game crash when attempting to load a world, though - about 6 seconds after the loading screen starts, my GPU hangs, first evidenced by my inability to scroll a web browser on my other screen, after which my mouse cursor also stalls. 

About 12 seconds after that starts, the game crashes. 

The crash log is as follows:

https://gist.github.com/SolarLune/205f98300b60908ad06f206999320bc6

I tried a few times more - setting the graphics quality settings to Max Performance didn't help; it still hung the GPU driver. Changing it from my desktop resolution (1440p) to 720p and turning off FSR2 as antialiasing didn't help either - it hung for a longer period of time this time, and then crashed Wayland (I presume) and kicked me out to the login prompt.

**Comment by Aftermath** [comment]
```
System:
  Kernel: 6.6.13-200.fc39.x86_64 arch: x86_64 bits: 64 compiler: gcc
    v: 2.40-13.fc39 Desktop: KDE Plasma v: 5.27.10 Distro: Fedora release 39
    (Thirty Nine)
CPU: AMD Ryzen 9 7950X3D 16-Core @ 32x 5.759GHz
Graphics:
  Device-1: AMD Navi 31 [Radeon RX 7900 XT/7900 XTX] vendor: Sapphire NITRO+
    driver: amdgpu v: kernel arch: RDNA-3 bus-ID: 03:00.0
  Display: wayland server: X.org v: 1.20.14 with: Xwayland v: 23.2.4
    compositor: kwin_wayland driver: X: loaded: amdgpu
    unloaded: fbdev,modesetting,radeon,vesa dri: radeonsi gpu: amdgpu
    resolution: 1: 2560x1440 2: 2560x1440 3: 2560x1080 4: 2560x1440
  API: EGL v: 1.5 drivers: radeonsi,swrast platforms:
    active: wayland,x11,surfaceless,device inactive: gbm
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 23.3.3 glx-v: 1.4
    direct-render: yes renderer: AMD Radeon RX 7900 XTX (radeonsi navi31 LLVM
    17.0.6 DRM 3.54 6.6.13-200.fc39.x86_64)
  API: Vulkan v: 1.3.268 drivers: radv,llvmpipe surfaces: xcb,xlib,wayland
    devices: 2
```

I was able to start a single player game running Proton Experimental: Bleeding Edge.
I also had to turn off a particular graphical setting regarding some voxel rendering option in order to have normal frames, otherwise it would render less than 1 FPS.

I do still get an crash of the display driver / display environment if I attempt to join a dedicated server.

**Comment by pollux78** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7442#issuecomment-1910828896

That seems to be the same problem for me, hosting works tho

**Comment by Blisto91** [comment]
Made mesa issue for the hang when entering a world https://gitlab.freedesktop.org/mesa/mesa/-/issues/10500

Edit: reworded as it likely isn't the intro video or entering the world for the first time part that is the cause.

**Comment by Aftermath** [comment]
FWIW, I updated to a kernel-vanilla/stable 6.7.1 from Fedora copr, and the bad behavior changed.
I can now connect to dedicated servers and play mostly normally, and Voxel Detail Models can be enabled now without killing frames.

Point Light Shadows are now broken though, and cause really low frames if enabled.
Otherwise, runs great.

Worth noting that there were several hotfixes targeting the game today. I don't know if any of those would have had a direct effect on our issues.

**Comment by SolarLune** [comment]
Back with another finding. As per ProtonDB, starting the game and loading into a single-player world works if I have the RADV drivers installed. On EndeavourOS / Arch, this is done through installing `vulkan-radeon` and `vulkan-icd-loader`, I believe, and then running the game with the command `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json:/usr/share/vulkan/icd.d/radeon_icd.i686.json %command%`. 

Also note that the game runs fine on Steam Deck at about 25-30 FPS on all Performance settings.

**Comment by TobiasLainson** [comment]
I'm also having similar problems, the game launches fine but when I join a multiplayer game it starts to load then I get a black screen, then loading screen again and after that the whole system hangs and I have to pull the power to restart computer.

I have also found a way to circumvent this hang, if I start a single player game first and then quit to start screen I am able to join a multiplayer game without the system hanging. Doing it this way I loose about 2/3 of my fps though. going from 60-100 to 19-25, mangohud is telling me my GPUs memory frequency is going down from a locked 1000Mhz in the single player game to a fluctuating 400-600Mhz during multiplayer.

System:
Nobara 39
Mesa 23.3.3
Kernel 6.7.0
Proton, its the same behavior with proton 8, experimental and ge-proton-8-30

**Comment by intelligentgaming** [comment]
The game runs fine on Ubuntu 23.10 with a RTX 3080 and Ryzen 7 5700X, even at some times beating Windows 11 performance on the same machine. 

However the only glitch I have found is that sometimes when launching the game, it will launch in windowed rather than full screen, but this can be easily fixed by toggling in game. 

Whether this is the game, Proton or Gnome is anyone's guess.

**Comment by Zepticon** [comment]
I have similar issues. In order to be able to join a multiplayer dedicated server, i need to sett my settings to "Max Performance".  When in game again, i turn them on one by one.

The following options causes issues for me:

- Contact Shadows: Crashes the game. Sometimes i get a crash message, sometimes i need to kill the computer.
- Reflections: Crashes the game. Sometimes i get a crash message, sometimes i need to kill the computer.
- Point Light: Kills frame rates. I drop to 2-3 fps on my 7900XTX

I can have all other settings on max without issues.

System:
Kubuntu on kernel 6.7.6
AMD 7800X3D
Sapphire RX 7900XTX Nitro+
64Gig 6000mhz memory

Edit:
It seems with those 4 settings disabled, I am able to launch the game as is. The strange thing is that sometimes i can enable some of the settings without it crashing. But not always. But i have steadily played with these off now for a two weeks and no more crashes.

Edit 2: 
I can also reliably launch the game with Voxel detail models set to max now. It crashes very rarely with this setting on.
Still not possible to launch with the above 3 settings on anything than low/off.

**Comment by AQDuck97** [comment]
The game feels very stuttery for me on Wayland as VRR is not being used even when forced to always on in Plasma settings. The game only has windowed or borderless mode and trying it in windowed mode then using KWin to fullscreen it (which enables VRR with every other program, you can tell by slight flickering) does not work for Enshrouded.

**Comment by pollux78** [comment]
with the newest patch it crashes a couple of times when trying to load the shader cache and trying to load into a world will stay in the loading screen for a couple of minutes then it kicks you back out to the menu, also some images are broken 

this was under latest proton experimental(bleeding edge)

rx 6700 mesa 24.0.3 kernel 6.8

![Screenshot_20240327_040610](https://github.com/ValveSoftware/Proton/assets/81181814/a172f77f-3ac5-4ba3-b53b-13ec3189a7f9)
![Screenshot_20240327_040555](https://github.com/ValveSoftware/Proton/assets/81181814/f605abbe-75d1-4b15-add3-c1ee9f94bdcf)

EDIT: latest proton experimental(bleeding edge) the game is now working properly, able to load in and main menu looks correct!

![Screenshot_20240328_032505](https://github.com/ValveSoftware/Proton/assets/81181814/af9c055a-0da3-44d0-b418-2216ef2c2285)

**Comment by bentalexanderhaase** [comment]
Can confirm, the remaining issue is the never ending 'validation' of around 10GB. Any known fix for this?

**Comment by urbenlegend** [comment]
> Can confirm, the remaining issue is the never ending 'validation' of around 10GB. Any known fix for this?

Yeah I am getting the same issue. 9.2 GB of data is being downloaded every time. I have a feeling it is due to the shader cache.

**EDIT:** Yeah I left Steam to finish processing the shaders in the background and now I no longer get the 9.2 GB download.

**EDIT 2:** Nvm, it still seems to trigger big downloads but only after a day or so. Doesn't happen with every launch of Steam.

**Comment by urbenlegend** [comment]
Another issue I am experiencing is if I set Texture settings to Max Quality, all the textures appear black. If I drop it down by one to Quality, it seems okay.

**Comment by kisak-valve** [comment]
Hello @urbenlegend, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by urbenlegend** [comment]
@kisak-valve Here's the proton logs: [steam-1203620.log](https://github.com/ValveSoftware/Proton/files/15490989/steam-1203620.log)

To generate these logs, I booted into the game, set the texture quality from Quality to Max Quality to trigger the black texture bug, and then loaded into my world. After spinning the camera around for a few seconds, I immediately quit the game.

Here are my [Steam Runtime Diagnostics](https://gist.github.com/urbenlegend/70fef3b36f2bf7a05d83385fb8fd7b14)

For anyone curious, this is what it looks like in game:

https://github.com/ValveSoftware/Proton/assets/241316/6a807e26-a56a-49ea-9d8e-3aec84a4c5e0

**Comment by esullivan-nvidia** [comment]
> Another issue I am experiencing is if I set Texture settings to Max Quality, all the textures appear black. If I drop it down by one to Quality, it seems okay.

Thank you for reporting this issue. I was able to determine this is an NVIDIA driver regression only present in r555. A fix will be shipped in a future driver release. If you wish to reference this bug in the future it is being tracked internally with bug number 4675675.

**Comment by pollux78** [comment]
with the latest enshrouded update the studio improved steamdeck support and added fsr3 which is working perfectly under proton experimental(bleeding edge)

the game also doesnt crash or freeze when switching around graphics settings so there are no more problems that i am seeing now, this under my rx 6700, mesa 24.1.4, kernel 6.10, cachy os

**Comment by faandg** [comment]
As of last week something broke when running the enshrouded **gameserver** in proton, should this be tracked here as well? See https://github.com/ValveSoftware/Proton/issues/7995 (it does not appear to be limited to Fedora) and the linked ticket to find some additional testing.

**Comment by ckupe** [comment]
FWIW, I use ProtonGE 9.10-9.12 on my desktop, I use Nvidia 555 on Fedora 40 Workstation (GNOME) and the actual game works just fine.

Steam (native)
ProtonGE 9.12

![image](https://github.com/user-attachments/assets/083a5bcd-7058-4ebf-9d28-61ab346ef5b8)

but yes, the enshrouded dedicated server in a container with proton is broken.

**Comment by kisak-valve** [comment]
One out of two of my steam deck crashes even tho they mirror each other for settings

Issue transferred from https://github.com/ValveSoftware/SteamOS/issues/1664.
@xJackxWickx posted on 2024-10-09T19:33:12:

#### Your system information

* Steam client version:
* SteamOS version: 
* Opted into Steam client beta?: [Yes/No] 
* Opted into SteamOS beta?: [Yes/No] 
* Have you checked for updates in Settings > System?: [Yes/No]

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen.

#### Steps for reproducing this issue:

1. Run the game enshrouded and it crashes at various points in the loading process. I realize it’s not really meant for the steam deck but like I said I have TEO steam decks that mirror each other and one for whatever reason cannot load.
2.  I’ve exhausted most means based off of other peoples similar issues. Nothing has worked. Uninstalled the game at least 35 times. I’ve installed cryo utilities and still nothing.
3. What I don’t understand is how my one steam deck runs fine with the bare minimum tweaks. Helllp

**Comment by pwasiewi** [comment]
I use kernel 6.6.3, mesa 23.3.0 and amd fury r9 card and the game crashes after 1 minute with proton 9.03, the system is frozen and reset is needed. I also used proton experimental, ge-20 and got even more crashes. Many steam games work very stable with my configuration. I even do not remember such dreadful crashes with them. They were maybe 5 years ago. Now if a game starts running it continues to work in my system. Thus, crashes are a huge problem of this game.

**Comment by pollux78** [comment]
> I use kernel 6.6.3, mesa 23.3.0 and amd fury r9 card and the game crashes after 1 minute with proton 9.03, the system is frozen and reset is needed. I also used proton experimental, ge-20 and got even more crashes. Many steam games work very stable with my configuration. I even do not remember such dreadful crashes with them. They were maybe 5 years ago. Now if a game starts running it continues to work in my system. Thus, crashes are a huge problem of this game.

Please provide logs with the crash if you can 

By doing PROTON_LOG=1 %command% as your launch command, it will output a file in your home directory to upload here 

You can also use something like qjournalctl to look at your last boot log and it should show what error made that happen

**Comment by pwasiewi** [comment]
[steam-1203620.log](https://github.com/user-attachments/files/17994366/steam-1203620.log)
My log and journal log https://gist.github.com/pwasiewi/bb087999f0e1a3c7ce75141a3150821c

**Comment by kisak-valve** [comment]
Hello @pwasiewi, the version of mesa you're using is too old to have https://gitlab.freedesktop.org/mesa/mesa/-/commit/79cb8842753e6dc7c10d6c5af1f1cf7ece5ecc75 included. Run the game with the launch options set to `RADV_DEBUG=zerovram %command%`.

**Comment by pwasiewi** [comment]
The same crash with RADV_DEBUG=zerovram %command% I will update mesa and come back with log.

**Comment by pwasiewi** [comment]
Updated mesa with launch option: PROTON_LOG=1 RADV_DEBUG=zerovram %command% and the same crash with log: 
[steam-1203620.log](https://github.com/user-attachments/files/17995751/steam-1203620.log)

**Comment by runar-work** [comment]
@pwasiewi You said you used kernel 6.6.3, but in your logs it says you're on 6.11.3-zen1. There's a kernel bug introduced in 6.10 which breaks the zerovram workaround (as described here: https://gitlab.freedesktop.org/drm/amd/-/issues/3812), so that could be why it still crashed for you with RADV_DEBUG=zerovram.

I tested Mesa 24.2.7 (and later 24.3.1) on kernels before and after the regression with RX 6650 XT to make sure it wasn't a new type of hang. I got four hangs with the newer kernel in four hours total, and none with the older kernel in three hours total. My repro rate wasn't great, but I would've expected at least one hang in the time I tested on the older kernel if it was a new/separate issue, so it seems likely that this is the same issue as before.

Until there's a fix for the kernel bug, the game should be fine on a 6.6 kernel (with Mesa 24.0.8 or newer, or RADV_DEBUG=zerovram).

**Comment by kisak-valve** [comment]
Enshrouded graphic problems [Pop!OS, Intel Arc a750]

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8430.
@zottelmann posted on 2025-01-31T11:52:29:

# Compatibility Report
- Name of the game with compatibility issues: Enshrouded
- Steam AppID of the game: 1203620

## System Information
- GPU: Intel Arc a750 Limited Edition
- Video driver version: Mesa Intel(R) Arc(tm) A750 Graphics (DG2) (0x56a1)  Version: 24.0.3

- Kernel version: Linux pop-os 6.9.3-76060903-generic
- Link to full system information report as [Gist](https://gist.github.com/):https://gist.github.com/zottelmann/36f93106fc90f06791480b440aed322b
- Proton version: Proton Hotfix/Experimental

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.



## Symptoms 
Game Starts, but a lot of  flickering graphics, missing textures, missing character models (PC, Enemies...) low performance
Tried different Proton Versions, reinstalled Mesa drivers, updated everything. No luck.


## Reproduction
Everytime playing the game



Cheers,
***
@mbriar commented on 2025-01-31T12:52:44:

This is a native vulkan game and there is no graphic api translation happening in proton. This is almost assuredly a Intel driver bug and should be reported to Mesa.

**Comment by kisak-valve** [comment]
Hello @zottelmann, if possible, please retest with a current build of mesa from a PPA like [kisak-mesa](https://launchpad.net/~kisak/+archive/ubuntu/kisak-mesa), and if it continues to misrender, then report your findings to your video driver vendor.

Possibly related to https://gitlab.freedesktop.org/mesa/mesa/-/issues/12387.

**Comment by zottelmann** [comment]
Hey thank you i will report the Problem to Mesa then.

**Comment by birdhimself** [comment]
One of the few games I couldn't get to work on the Radeon 9070 XT so far. Works fine with the Radeon 7800 XT with the same setup.

Kernel 6.13.5
Mesa 25.0.0
kernel-firmware-amdgpu-20250302-1.1 (openSUSE Tumbleweed package)

Crashes with this:

```
[W 00:00:00,574] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[W 00:00:00,585] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[W 00:00:00,602] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[W 00:00:00,618] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[I 00:00:00,642] [app] start creation step Preload
[W 00:00:00,684] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[E 00:00:00,696] [graphics] Failed to allocate download buffer memory.
[W 00:00:00,698] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[E 00:00:00,701] [graphics] Failed to allocate download buffer memory.
[W 00:00:00,705] [graphics] vkQueuePresentKHR failed with error 'VK_SUBOPTIMAL_KHR'
[I 00:00:00,902] [online] User stats received.
[E 00:00:10,863] [graphics] vkWaitForFences() timed out after 10.00 s.. probably a driver crash!
[E 00:00:10,863] [graphics] Vulkan device lost...
[I 00:00:10,863] [DeviceFaultEXT] A GPUVM fault has been detected
[I 00:00:10,864] [DeviceFaultEXT] Caused by 1 in address range [ 18446603339462164480, 18446603339462168575 ]
[E 00:00:10,864] Stopping process due to unrecoverable GPU crash!
```

Tried Proton Experimental and bleeding-edge.

**Comment by linde12** [comment]
@C0Nd3Mnd there is a merge request for this referenced here: https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/34027

I tested it and it works with the 9070 XT, but for now you have to build mesa from source (not released or even merged to main yet)

**Comment by SemDyal** [comment]
The game stays locked on the Preprocessing screen. The log indicates a failure to allocate memory but my RAM has still plenty of space according to the task manager.

I tried to launch the game with Proton 9.0-4, Proton Experimental and GE-Proton9-27.

Device: AMD Radeon RX 590 Series (radeonsi, polaris10, ACO, DRM 3.59, 6.8.0-57-generic) (0x67df)
Kernel: 6.8.0-57-generic
Mesa: 25.0.3 - kisak-mesa PPA

[enshrouded_2025-04-09T11-16-05.log](https://github.com/user-attachments/files/19664158/enshrouded_2025-04-09T11-16-05.log)

EDIT: Retried today, after some updates of the system. Works with Proton 9.0-4.

**Comment by Galcian79** [comment]
I have a different issue where the gpu hangs randomly. 
It doesn't happens on every play, but only when loading a save. I have included a trace but umr logs are empty, probably because the gpu resets before it could take a trace.

### Software information
Archlinux

### System information
- GPU: Radeon RX 6700 XT
- Driver: Mesa 1.25.0.3 and GIT (the trace one)
- Proton version: experimental_be

### Log files

[steam-1203620.tar.gz](https://github.com/user-attachments/files/19670509/steam-1203620.tar.gz)

[radv_dumps_1556046_2025.04.09_18.11.06.tar.gz](https://github.com/user-attachments/files/19670512/radv_dumps_1556046_2025.04.09_18.11.06.tar.gz)

[dmesg.log](https://github.com/user-attachments/files/19670519/dmesg.log)

**Comment by KarlVogel** [comment]
Not sure if this is a proton issue, but noticed the fossilize replay taking ages for this game (also saw people mentioning in reviews that shader precaching was taking a long time).

Noticed that the shadercache file is enormous:

```
$ stat -c '%s' .local/share/Steam/steamapps/shadercache/1203620/fozpipelinesv6/steam_pipeline_cache.foz
13580667237
```

13.5Gb cache file.

```
$ fossilize-list .local/share/Steam/steamapps/shadercache/1203620/fozpipelinesv6/steam_pipeline_cache.foz --tag 4 --size|sort -n -k 2|tail -4
6037bf64a9fe6952 3335367 compressed bytes, 8025120 uncompressed bytes
5087898493216e29 3343291 compressed bytes, 8031903 uncompressed bytes
234b5e467d71bda2 3344574 compressed bytes, 8029806 uncompressed bytes
41146b3da0507261 3344596 compressed bytes, 8030639 uncompressed bytes
```

Seems it includes the entire shader hlsl source code a couple of times. Probably an issue with the game, but maybe something that could be more efficient in proton too or going wrong? Not too familiar with this stuff.

ie. snippit like this is present 14x times, in the disasm output

```
$ fossilize-disasm .local/share/Steam/steamapps/shadercache/1203620/fozpipelinesv6/steam_pipeline_cache.foz --filter-module 41146b3da0507261 --output /tmp/ --device-index 0

...
  47   │ OpSource HLSL 620 %21 "#line 1 \"E:/keen/checkouts/Default//holistic/content/shader/platform.hlsl\"
  48   │ 
  49   │ #ifndef KEEN_PLATFORM_HLSL_INCLUDED
  50   │ #define KEEN_PLATFORM_HLSL_INCLUDED
  51   │ 
  52   │ // rebuild counter: 10
  53   │ 
  54   │ #ifndef KEEN_CONCATENATE
  55   │ #   define KEEN_CONCATENATE_HELPER( value1, value2 ) value1 ## value2
  56   │ #   define KEEN_CONCATENATE( value1, value2 ) KEEN_CONCATENATE_HELPER( value1, value2 )
  57   │ #endif
  58   │ 
  59   │ #define KEEN_IDENTITY( value )              value
  60   │ #define KEEN_MAKE_VARNAME( prefix )         KEEN_CONCATENATE( prefix, KEEN_IDENTITY( __LINE__ ) )
  61   │ #define KEEN_STATIC_ASSERT( EXPRESSION )    typedef uint KEEN_MAKE_VARNAME( keen_static_assert ) [ (EXPRESSION) ? 1 : -1 ]
  62   │ 
  63   │ #if defined( KEEN_FLOAT16 )
  64   │ typedef float16_t float16;
  65   │ typedef float16_t2 float16_2;
  66   │ typedef float16_t3 float16_3;
  67   │ typedef float16_t4 float16_4;
  68   │ #endif
  69   │ 
  70   │ #if defined( KEEN_PLATFORM_PS5 )
  71   │ #   define KEEN_STRUCT_BUFFER           RegularBuffer
...
```

**Comment by kisak-valve** [comment]
Enshrouded Crash on Client Connect Proton 10

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8777.
@Maylos posted on 2025-06-06T09:44:10:

# Compatibility Report
- Name of the game with compatibility issues: Enshrouded
- Steam AppID of the game: 1203620

## System Information
- GPU: RX 9070 XT
- Video driver version: Mesa 25.1.2
- Kernel version: 6.14.9
- Link to full system information report as [Gist](https://gist.github.com/): [https://gist.github.com/Maylos/aeea1d6b229580dd91ea724dba39c977](https://gist.github.com/Maylos/aeea1d6b229580dd91ea724dba39c977) 
- Proton version: Proton 10.0-1

## I confirm:
- [Yes but not this specific issue.] that I haven't found an existing compatibility report for this game.
- [Yes ] that I have checked whether there are updates for my system available.



[steam-1203620.log](https://github.com/user-attachments/files/20625554/steam-1203620.log)

## Symptoms 

Connecting to any dedicated server or co-op partner results in a crash as soon as loading screen appears. 
Hosting a game or playing Solo works as expected. 
Working as expected with 9.0-4 and running Enshrouded_Server.exe with Proton 10.0-1 also works as expected.
Appears to only be an issue when connecting to another host.  

## Reproduction

Connect to any dedicated server or co-op partner when running with Proton 10.0-1

**Comment by Kawazu0** [comment]
successfully connected  on 2 servers with proton experimental-10.0-20250612c

regards

**Comment by Maylos** [comment]
> successfully connected on 2 servers with proton experimental-10.0-20250612c
> 
> regards

Thanks heaps! Can confirm working as expected now with Proton Experimental.

**Comment by PaulSuarez1** [comment]
Linux/AMD workaround for Enshrouded not launching: jemalloc fix

Posting this because I just spent an embarrassing amount of hours debugging this and I don't want anyone else to go through the same thing.

I'm on Linux with a 7900 XTX running Enshrouded through Proton. Game was working totally fine for almost 3 hours, had a crash mid-session, and after that it just refused to launch. Hit Play, it thinks for a few seconds, then the button goes back to green. No error window, no crash popup, just... nothing.

I tried everything:
- Multiple Proton versions (GE-Proton, Experimental, Hotfix, 9.0-4, 10.0-4)
- Nuking the prefix over and over
- Clearing shader caches
- Verifying game files
- Deleting the game config
- Full reinstall
- Reinstall to a completely different drive
- Resetting Steam Linux Runtime Sniper (which actually did fix a secondary issue — 8,248 files had to be reacquired)
- Rebooting
- Forcing GPU selection with MESA_VK_DEVICE_SELECT
- Blocking NvLowLatencyVk.dll through Wine DLL overrides

None of it worked.

What actually fixed it: swapping the memory allocator to jemalloc.

---

**How to tell if you have the same problem**

Set your launch options to this and try launching once:

    PROTON_LOG=1 RADV_PERFTEST=gpl %command%

Let it fail, then check ~/steam-1203620.log and search for:

    malloc(): unaligned tcache chunk detected

If that line is in there, you're probably dealing with the same thing.

Your game log (enshrouded.log in the game directory) will cut off somewhere around Vulkan device setup, usually near a line about DLSS.

---

**The fix**

Install jemalloc for your distro:
- Arch: `sudo pacman -S jemalloc`
- Fedora: `sudo dnf install jemalloc`
- Ubuntu/Debian: `sudo apt install libjemalloc2`
- NixOS: add jemalloc to your systemPackages or use nix-shell

Find where libjemalloc.so ended up:

    find /usr -name "libjemalloc.so*" 2>/dev/null

Then set your Enshrouded launch options to:

    LD_PRELOAD=/path/to/libjemalloc.so RADV_PERFTEST=gpl %command%

For Arch that's usually:

    LD_PRELOAD=/usr/lib/libjemalloc.so RADV_PERFTEST=gpl %command%

For Fedora:

    LD_PRELOAD=/usr/lib64/libjemalloc.so RADV_PERFTEST=gpl %command%

---

**Why this works (best I can tell)**

Enshrouded ships with NvLowLatencyVk.dll, which is NVIDIA's Reflex Vulkan layer. On AMD hardware under Proton, this DLL still loads and hooks into Vulkan initialization even though it doesn't do anything useful. During Vulkan startup, Wine's vulkan-1.dll goes through an unload/reload cycle, and the Reflex hooks seem to cause some heap corruption in that process. glibc catches it and kills the process before the game ever gets to the menu.

With jemalloc handling memory allocation instead, the game gets past that init sequence, properly detects "non-Nvidia GPU," skips DLSS, creates the Vulkan device, and runs normally. I played for a few hours after applying this with zero issues. jemalloc is used by Firefox, Redis, and a bunch of other production software so it's not some sketchy hack.

I want to be clear, I'm not 100% certain this is the definitive root cause. But I am certain that this is the workaround that got it running after testing literally every other option. The malloc tcache error is consistent and reproducible, and jemalloc consistently gets past it.

---

**Note for the devs if you see this**

The game already correctly detects non-NVIDIA GPUs later in init (there's a log line that says "Skipping DLSS on non-Nvidia GPU.") If that vendor check happened earlier and prevented NvLowLatencyVk.dll from initializing its Vulkan hooks on AMD hardware, this whole class of crash probably goes away for Linux AMD users. Making the Reflex DLL a soft dependency or gating its load behind a GPU vendor check would be a clean fix.

Also worth noting: once you're in game on AMD, set NVIDIA Reflex to Off in the settings. It does nothing on AMD hardware and it's tied to the DLL that causes the problem in the first place.

---

**Setup for reference**

AMD Ryzen 9 / RX 7900 XTX / Mesa RADV 26.x / kernel 6.12 LTS / tested across GE-Proton, Proton Experimental, Proton Hotfix, Proton 9.0-4, and Proton 10.0-4. All same crash, all fixed by jemalloc.

Hope this helps somebody. If you see that malloc tcache line in your Proton log, give jemalloc a shot before you pull your hair out.

**Comment by kisak-valve** [comment]
Enshrouded (1203620) — Unknown IPC call assertions for IClientUserStats/IClientRemoteStorage/IClientHTTP after update, "Steam client not running" dialog

Issue transferred from https://github.com/ValveSoftware/Proton/issues/10120.
@flying-disk posted on 2026-09-04T22:12:57:

Compatibility Report, compiled by Claude from my troubleshooting chat.

[steam-1203620.log](https://github.com/user-attachments/files/31853332/steam-1203620.log)

[enshrouded.log](https://github.com/user-attachments/files/31853339/enshrouded.log)

Name of the game: Enshrouded
Steam AppID: 1203620

System Information
- GPU: NVIDIA RTX 3060, driver 595.84.0
- Distro: Linux Mint 22.3
- Proton versions tested: 9.0, 10.0, 11.0, Proton Experimental, Proton Hotfix, GE-Proton (latest) — all fail identically
- Steam client: tested both stable and Beta — same result

Symptoms
Game was working normally when last played two days ago. Played a different game yesterday, then today the launch fails with the game's own "Your Steam client is not running!" dialog, despite SteamAPI_Init() succeeding. All other installed games work normally.

Proton log (steam-1203620.log) shows:
fixme:steamclient:manual_convert_Set_SteamAPI_CCheckCallbackRegisteredInProcess_func_156 not implemented!

Steam client log shows repeated assertion failures right after game launch:
src/clientdll/interfacemap_client.h (5617): Unknown IPC call IClientNetworkingSocketsSerialized::0xaf9f21ff
src/clientdll/interfacemap_client.h (3534): Unknown IPC call IClientRemoteStorage::0x8694e986
src/clientdll/interfacemap_client.h (3897): Unknown IPC call IClientHTTP::0xe12de80b
src/clientdll/interfacemap_client.h (3133): Unknown IPC call IClientUserStats::0x731946ee

Associated Valve crash-report IDs from these assertions:
bp-006f285a-c768-4229-a928-054fa2260904
bp-5f791c99-0630-47a6-9b0e-c0e4f2260904
bp-fce1591c-de88-4e20-959f-117d52260904
bp-0776eccf-e414-4cb5-9f3f-bf5042260904

Enshrouded's own log confirms the game-side failure point:
[E] [online] Could not get current user stats.
[E] [achievement] Could not create provider. Error: internal
[X] FATAL ERROR: Online_InitializationFailed

This suggests lsteamclient's interface map doesn't yet recognize interface
versions Enshrouded's latest build calls for IClientUserStats, IClientRemoteStorage,
IClientHTTP, and IClientNetworkingSocketsSerialized.

Reproduction: 100% reproducible on every launch attempt today.

**Comment by masives** [comment]
I have a different issue that I couldn't find any answer to. Game loads, loads world and in most places it works normally. However, when I interact with tower(spire?) then it will start loading and never finish.  In logs I managed to find this which seems to be the issue:

```
Player EntityId 1 teleported From (3508.231934, 921.000000, 1956.755981) to (3504.750000, 973.000000, 1969.750000)
[I 00:00:39,678] [rendering] pending scene render finished flags: { ModelLoader, VoxelMeshing }
[I 00:00:39,817] [rendering] pending scene render finished flags: { GpuCache, ModelLoader, VoxelMeshing }
[I 00:00:39,851] [rendering] pending scene render finished flags: { ModelLoader, VoxelMeshing }
[I 00:00:40,197] [rendering] pending scene render finished flags: { GpuCache, ModelLoader, VoxelMeshing }
[I 00:00:40,231] [rendering] pending scene render finished flags: { ModelLoader, VoxelMeshing }
[I 00:00:41,299] [rendering] pending scene render finished flags: { ModelLoader }
[I 00:01:01,931] -------------- Session ----------------
[I 00:01:01,931] Machines:
[I 00:01:01,931]   m#0(128): up 0 (0), down 0 (0), remote 0 (0), limit 256, lost 0, ping 0 ms, EstablishingBaseline
[I 00:01:01,931] ---------------------------------------
[I 00:01:31,958] -------------- Session ----------------
[I 00:01:31,958] Machines:
[I 00:01:31,958]   m#0(128): up 0 (0), down 0 (0), remote 0 (0), limit 256, lost 0, ping 0 ms, EstablishingBaseline
[I 00:01:31,958] ---------------------------------------
[I 00:02:01,961] -------------- Session ----------------
[I 00:02:01,961] Machines:
[I 00:02:01,961]   m#0(128): up 0 (0), down 0 (0), remote 0 (0), limit 256, lost 0, ping 0 ms, EstablishingBaseline
[I 00:02:01,961] ---------------------------------------
[I 00:02:16,360] [rendering] pending scene render finished flags: { ModelLoader, Weather }
[I 00:02:16,391] [rendering] pending scene render finished flags: { ModelLoader }
[E 00:02:28,897] [gpu_data_cache] failed to allocate 177,152 bytes of device memory for gpu texture '' (free: 32,642): out of memory
[E 00:02:28,897] [gpu_data_cache] failed to allocate 352,256 bytes of device memory for gpu texture '' (free: 32,642): out of memory
[E 00:02:28,897] [gpu_data_cache] failed to allocate 46,080 bytes of device memory for gpu texture '' (free: 32,642): out of memory
[E 00:02:28,897] [gpu_data_cache] failed to allocate 92,160 bytes of device memory for gpu texture '' (free: 32,642): out of memory
```

`Enshrouded will not finish loading due to could not allocate 92160 bytes of texture data (free: 41858 bytes)` 

So games works normally until it teleports me somewhere and starts to load larger texture then it just stops trying or clear the vram up? It's the only game I had issue with. 

pop_os 24:04, Proton Hotfix, AMD Ryzen 5 5600X (12) @ 4.654GHz, AMD ATI Radeon RX 470, Mesa 26.1.6
